docs/commands/analyze.md

```diff
@@ -0,0 +1,54 @@
+---
+parent: Commands
+title: modular analyze
+---
+
+# `modular analyze <package-name>`
+
+Analyzes the dependencies of a package from its source code, emitting JSON to
+`stdout` (please note that errors and warnings can still be emitted to
+`stderr`).
+
+## Output
+
+The output of the command is a JSON object, with a `resolution` and a `manifest`
+field, containing the respectively computed dependency versions for the
+specified package. For example:
+
+```json
+{
+  "manifest": {
+    "react": "^17.0.2",
+    "@mantine/core": "^3.6.13",
+    "lodash": "^4.17.21",
+    "lodash.merge": "^4.6.2",
+    "ress": "^5.0.2"
+  },
+  "resolutions": {
+    "react": "17.0.2",
+    "@mantine/core": "3.6.14",
+    "lodash": "4.17.21",
+    "lodash.merge": "4.6.2",
+    "ress": "5.0.2"
+  }
+}
+```
+
+## How versions are calculated
+
+The source code of the specified package is statically analyzed to find all the
+`import` or `require` statements. Then, for each external dependency imported in
+the code, a its `manifest` version is computed by looking at the workspace's
+manifest files, and its `resolution` version is computed by parsing the
+workspace's `yarn.lock`.
+
+For this reason, `resolution` versions are strictly exact, while `manifest`
+versions can contain any ammissible type of
+[ranged semver](https://github.com/npm/node-semver#versions). `manifest`
+versions are set to the first successful lookup from the following sequence of
+lookup locations::
+
+1. The package's `package.json` `dependencies` field
+1. The package's `package.json` `devDependencies` field
+1. The root `package.json` `dependencies` field
+1. The root `package.json` `devDependencies` field
```

Comment:
This numbered list only has `1.`

Reply:
I thought this section was to specify the order in which you perform the look up for dependency versions. Wouldn't it make sense to make it an ordered list so it's clear the order of logic?

Reply:
> Wouldn't it make sense to make it an ordered list so it's clear the order of logic?

This is an ordered list, and its order is implicit in the sequential ordering in the text. Users will see it rendered with the "correct" numbers. If you use the same number, it's just easier (and less prone to errors) to add more steps in the middle with just one line change in the future. That's why I normally use 1. everywhere (unless I have to reference the lines by number in the text).

Anyway, it makes little difference :) If you feel explicit numbers are better, I'll change them.

Reply:
Done!

Reply:
Ahhh I see I see. This is a markdown implementation. You can ignore :)

Reply:
> Ahhh I see I see. This is a markdown implementation. You can ignore :)

I just changed them 🤣